test/Ocelot.UnitTests/Consul/ProviderFactoryTests.cs

```diff
@@ -43,15 +38,15 @@ public void should_return_ConsulServiceDiscoveryProvider()
     }
 
     [Fact]
-    public void should_return_PollingConsulServiceDiscoveryProvider()
+    public void ShouldReturnPollingConsulServiceDiscoveryProvider()
     {
         var provider = DummyPollingConsulServiceFactory(string.Empty);
         var pollProvider = provider as PollConsul;
         pollProvider.ShouldNotBeNull();
     }
 
     [Fact]
-    public void should_return_SameProviderForGivenServiceName()
+    public void ShouldReturnSameProviderForGivenServiceName()
```

Comment:
Hmm... BDDfy can work with different name conventions: ["You can use underscored or pascal or camel cased method names for your steps"](https://github.com/TestStack/TestStack.BDDfy/blob/10ee0c95af78828116aac02d7283777a85256ef0/readme.md?plain=1#L7)
But the C# naming convention is PascalCase. As for C# developer, as for me, it is very strange to see C & C++ naming convention with `_` (underscores) in method names!
In future I am going to fix this huge problem.

**2nd** problem. Visual Studio generates a lot of IDE1006 `Naming rule violation` because method name is not PascalCase. 
But for now, it seems scenarios with PascalCase is not logged into build log at all. Probably it is Ok because passed scenario should not be logged by default.
Currently all BDDfy unit tests are logged producing huge logs... But this is another story.

It makes a lot of sense to change names from underscore-style (C++) to C# PascalCase because of the following:
- Passed unit test will not be logged by BDDfy framework to build console on CircleCI
- Method names of unit tests don't violate default [C# naming conventions](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/identifier-names#naming-conventions) (aka IDE1006 rule, etc.)

Reply:
@raman-m Ok, Raman it makes sense: So, we could keep it like that and create a new PR for theories / facts naming conventions. I'm not used to the underscore naming convention either.

Reply:
✔️ Fixed in https://github.com/ThreeMammals/Ocelot/pull/1711/commits/842a94135973c71c7770070936b2930f9ab16bc5

Reply:
It makes too much sense... But not in this PR which I would like to merge today :)
It will another huge PR to resolve mentioned problems... in October from me... 😸 

It seems I must create another follow up issue after this follow up... 🤣 I'm going crazy with this project...